uaclient/contract.py

```diff
@@ -68,30 +70,41 @@ def request_root_macaroon(self):
         self.cfg.write_cache('root-macaroon', root_macaroon)
         return root_macaroon
 
-    def request_accounts(self, macaroon_token):
+    def request_accounts(self, macaroon_token=None, machine_token=None):
         """Request list of accounts this user has access to."""
         headers = self.headers()
-        headers.update({'Authorization': 'Macaroon %s' % macaroon_token})
+        if macaroon_token:
+            headers.update({'Authorization': 'Macaroon %s' % macaroon_token})
+        else:
+            headers.update({'Authorization': 'Bearer %s' % machine_token})
         accounts, _headers = self.request_url(
             API_V1_ACCOUNTS, headers=headers)
         self.cfg.write_cache('accounts', accounts)
         return accounts
 
-    def request_account_contracts(self, macaroon_token, account_id):
+    def request_account_contracts(
+            self, account_id, macaroon_token=None, machine_token=None):
         """Request a list of contracts authorized for account_id."""
         url = API_V1_TMPL_ACCOUNT_CONTRACTS.format(account=account_id)
         headers = self.headers()
-        headers.update({'Authorization': 'Macaroon %s' % macaroon_token})
+        if macaroon_token:
+            headers.update({'Authorization': 'Macaroon %s' % macaroon_token})
+        else:
+            headers.update({'Authorization': 'Bearer %s' % machine_token})
         account_contracts, _headers = self.request_url(url, headers=headers)
         self.cfg.write_cache('account-contracts', account_contracts)
         return account_contracts
 
-    def request_add_contract_token(self, macaroon_token, contract_id):
+    def request_add_contract_token(
+            self, contract_id, macaroon_token=None, machine_token=None):
         """Create a contract token for use when adding a machine to a contract
 
         """
         headers = self.headers()
-        headers.update({'Authorization': 'Macaroon %s' % macaroon_token})
+        if macaroon_token:
+            headers.update({'Authorization': 'Macaroon %s' % macaroon_token})
+        else:
+            headers.update({'Authorization': 'Bearer %s' % machine_token})
```

Comment:
Same issue as above.

Reply:
(Given this appears in multiple places, a function which handles this logic, including the appropriate error messaging might make sense?)

Reply:
really dropped